### PR TITLE
feat(sync-actions): publish product actions

### PR DESCRIPTION
**Pull request by markus-azer**
#### Description

Give the ability to publish product actions directly by simply mark `staged` flag to false.

One use case for this, it would make it easy for package user to just publish the updates instead of sending update actions then send another action to publish product updates plus this feature would shorten the platform calls

### CODE REVIEW

File: packages/sync-actions/src/products.js [hunk 1]
```diff
@@ -129,6 +130,9 @@ function createProductMapActions(
       )
     )
 
+    if (publish === true)
+      return flatten(allActions).map((action) => ({ ...action, staged: false }))
```
Comment (daern91) [inline packages/sync-actions/src/products.js:134]: Will this also work for the [setKey](https://docs.commercetools.com/http-api-projects-products#set-key) and [setTaxCategory](https://docs.commercetools.com/http-api-projects-products#set-taxcategory) actions? 

I see they are in the list here
https://github.com/commercetools/nodejs/blob/2008434e12e469d4d035ed4658cff51069ba7985/packages/sync-actions/src/product-actions.js#L26
and here
https://github.com/commercetools/nodejs/blob/2008434e12e469d4d035ed4658cff51069ba7985/packages/sync-actions/src/product-actions.js#L36

but am not sure how it works.

Comment (markus-azer) [reply packages/sync-actions/src/products.js:134]: Thats good question, i thought its only `setkey` that doesnt have this flag and i already tested it and it worked, seems that platform ignored  this flag, i should also try it with `setTaxCategory `

File: packages/sync-actions/src/products.js [hunk 2]
```diff
@@ -43,6 +43,7 @@ function createProductMapActions(
   ): Array<UpdateAction> {
     const allActions = []
     const { sameForAllAttributeNames } = options
+    const { publish } = newObj
```
Comment (markus-azer) [inline packages/sync-actions/src/products.js:46]: regarding this flag naming, i thought that `publish` is more clear than using `staged`.

File: packages/sync-actions/src/products.js (hunk 1 above)
Comment (daern91) [reply packages/sync-actions/src/products.js:134]: Sounds good to me. And we don't have integration tests, right? So I guess it's all up to us to manually test it and make sure the platform doesn't throw on us.

Comment (markus-azer) [reply packages/sync-actions/src/products.js:134]: yep so far we dont have integration tests, and i just tried it with `setTaxCategory` and it worked fine

**Pull request merged**